GitUI/UserControls/RevisionGrid/Graph/Rendering/GraphRenderer.cs

```diff
@@ -0,0 +1,229 @@
+﻿using System.Drawing.Drawing2D;
+using GitCommands;
+using GitExtUtils.GitUI;
+using GitUIPluginInterfaces;
+
+namespace GitUI.UserControls.RevisionGrid.Graph.Rendering
+{
+    internal static class GraphRenderer
+    {
+        internal const int MaxLanes = RevisionGraph.MaxLanes;
+
+        public static readonly int LaneLineWidth = DpiUtil.Scale(2);
+        public static readonly int LaneWidth = DpiUtil.Scale(16);
+        public static readonly int NodeDimension = DpiUtil.Scale(10);
+
+        private const int _noLane = -10;
+
+        public static void DrawItem(Graphics g, int index, int width, int rowHeight,
+            Func<int, IRevisionGraphRow?> getSegmentsForRow,
+            RevisionGraphDrawStyleEnum revisionGraphDrawStyle,
+            ObjectId headId)
+        {
+            SmoothingMode oldSmoothingMode = g.SmoothingMode;
+            Region oldClip = g.Clip;
+
+            int top = g.RenderingOrigin.Y;
+            Rectangle laneRect = new(0, top, width, rowHeight);
+            using Region newClip = new(laneRect);
+            newClip.Intersect(oldClip);
+            g.Clip = newClip;
+            g.Clear(Color.Transparent);
+
+            DrawItem();
+
+            // Restore graphics options
+            g.Clip = oldClip;
+            g.SmoothingMode = oldSmoothingMode;
+
+            return;
+
+            void DrawItem()
+            {
+                IRevisionGraphRow? currentRow = getSegmentsForRow(index);
+                if (currentRow is null)
+                {
+                    return;
+                }
+
+                IRevisionGraphRow? previousRow = getSegmentsForRow(index - 1);
+                IRevisionGraphRow? nextRow = getSegmentsForRow(index + 1);
+
+                SegmentPointsInfo p = new();
```

Comment:
Structs can be initialised with `default`. E.g.,
```suggestion
                SegmentPointsInfo p = default;
```

Not requesting a change, just FYI.